pkg/ipam/crd.go

```diff
@@ -710,9 +716,15 @@ func (a *crdAllocator) AllocateWithoutSyncUpstream(ip net.IP, owner string) (*Al
 		return nil, err
 	}
 
+	result, err := a.buildAllocationResult(ip, ipInfo)
+	if err != nil {
+		log.WithError(err).WithField(logfields.IPAddr, ip).Warning("buildAllocationResult failed")
+		return nil, err
```

Comment:
I think instead of logging a warning which won't really be actionable to the user (how do they prevent it?), I think we can format the error msg instead so that we (as users) know the error is coming from this level of the code and to describe the problem further. From there, we can theorize why the IP might not be available in the CiliumNode:

```suggestion
		return nil, fmt.Errorf("failed to associate IP %s inside CiliumNode: %w", ip, err)
```

What that error msg is saying is that essentially we couldn't find the corresponding interface (ENI, Azure, Alibaba) in the local CiliumNode matching the IP `ip`.